src/wallet/init.cpp

```diff
@@ -39,7 +39,7 @@ const WalletInitInterface& g_wallet_init_interface = WalletInit();
 void WalletInit::AddWalletOptions() const
 {
     gArgs.AddArg("-addresstype", strprintf("What type of addresses to use (\"legacy\", \"p2sh-segwit\", or \"bech32\", default: \"%s\")", FormatOutputType(DEFAULT_ADDRESS_TYPE)), false, OptionsCategory::WALLET);
-    gArgs.AddArg("-avoidpartialspends", strprintf("Group outputs by address, selecting all or none, instead of selecting on a per-output basis. Privacy is improved as an address is only used once (unless someone sends to it after spending from it), but may result in slightly higher fees as suboptimal coin selection may result due to the added limitation (default: %u)", DEFAULT_AVOIDPARTIALSPENDS), false, OptionsCategory::WALLET);
+    gArgs.AddArg("-avoidpartialspends", strprintf("Group outputs by address, selecting all or none, instead of selecting on a per-output basis. Privacy is improved as an address is only used once (unless someone sends to it after spending from it), but may result in slightly higher fees as suboptimal coin selection may result due to the added limitation (default: %u%s)", DEFAULT_AVOIDPARTIALSPENDS, DEFAULT_AVOIDPARTIALSPENDS ? "" : ", but enabled for wallets with \"avoid_reuse\" enabled"), false, OptionsCategory::WALLET);
```

Comment:
Are you suggesting #ifdef? Because we might switch this default at some point.

Reply:
I just think it's weird to have runtime programming logic to construct a string where we already know what that string should be. If the concern is about this help text not being updated if `DEFAULT_AVOIDPARTIALSPENDS` changes, then I think it would be better to just leave a comment where that constant is defined saying "If this is changed to true, update the help text for -avoidpartialspends"

Reply:
I am going to just remove the conditional part of this and have it always say `"(always enabled for wallets with \"avoid_reuse\" enabled)"`. That should resolve the issue, I think?